Modules/vc-module-catalog/VirtoCommerce.CatalogModule.Test/ExportImport2Test.cs

```diff
@@ -0,0 +1,575 @@
+using System.Linq;
+using Moq;
+using Newtonsoft.Json;
+using VirtoCommerce.CatalogModule.Core.Model;
+using VirtoCommerce.CatalogModule.Core.Model.Search;
+using VirtoCommerce.CatalogModule.Core.Search;
+using VirtoCommerce.CatalogModule.Core.Services;
+using VirtoCommerce.CatalogModule.Data.ExportImport;
+using VirtoCommerce.CoreModule.Core.Conditions;
+using VirtoCommerce.ExportModule.Test.MockHelpers;
+using Xunit;
+
+namespace VirtoCommerce.CatalogModule.Test
+{
+    public class ExportImport2Test
+    {
+        [Fact]
+        public void CatalogFetchAll()
+        {
+            var catalogExportPagedDataSource = InitCatalogExportPagedDataSource();
+            var fetchResult = catalogExportPagedDataSource.FetchNextPage();
+            Assert.Equal(7, fetchResult.Count());
+        }
+
+        [Fact]
+        public void CatalogFetchForSpecificIds()
+        {
+            var catalogExportPagedDataSource = InitCatalogExportPagedDataSource();
+            catalogExportPagedDataSource.DataQuery.ObjectIds = new[] { "7829d35f417e4dd98851f51322f32c23" };
+            var fetchResult = catalogExportPagedDataSource.FetchNextPage();
+            Assert.Single(fetchResult);
+        }
+
+        private CatalogExportPagedDataSource InitCatalogExportPagedDataSource()
+        {
+            var authorizationServicesMock = AuthMockHelper.AuthServicesMock(true);
+            var сatalogSearchServiceMock = new Mock<ICatalogSearchService>();
+            сatalogSearchServiceMock.Setup(x => x.SearchCatalogsAsync(It.IsAny<CatalogSearchCriteria>())).ReturnsAsync(GetTestCatalogSearchResult());
+            var сatalogServiceMock = new Mock<ICatalogService>();
+            сatalogServiceMock.Setup(x => x.GetByIdsAsync(new string[] { "7829d35f417e4dd98851f51322f32c23" }, null)).ReturnsAsync(GetTestCatalogByIdsResult());
+            return new CatalogExportPagedDataSource(
+                сatalogSearchServiceMock.Object,
+                сatalogServiceMock.Object,
+                authorizationServicesMock.AuthorizationPolicyProvider,
+                authorizationServicesMock.AuthorizationService,
+                authorizationServicesMock.UserClaimsPrincipalFactory,
+                authorizationServicesMock.UserManager)
+            {
+                DataQuery = new CatalogExportDataQuery()
+            };
+        }
+
+        private CatalogSearchResult GetTestCatalogSearchResult()
```

Comment:
Could you reduce catalogs count to 2 and remove redundant collection properties like Properties, etc.?